### PR TITLE
stat/distuv: Fix Beta.Prob and Beta.LogProb for alpha = 1 and x = 0 or beta = 1 and x = 1

### LINKED ISSUE #1377: stat/distuv.Beta: Beta{Alpha: 1, Beta: 1}.Prob(0) == NaN
### What are you trying to do?
Computing the probability function of Beta(1,1) at various x,  0 <= x <= 1.

### What did you do?
distuv.Beta{Alpha: 1, Beta: 1}.Prob(0)

### What did you expect to happen?
I expected distuv.Beta{Alpha: 1, Beta: 1}.Prob(0) == 1

### What actually happened?
The result is NaN.

go version go1.14.2 darwin/amd64

> grep gonum.org/v1/gonum go.sum
gonum.org/v1/gonum v0.0.0-20180816165407-929014505bf4/go.mod h1:Y+Yx5eoAFn32cQvJDxZx5Dpnq+c3wtXuadVZAcxbbBo=
gonum.org/v1/gonum v0.7.0 h1:Hdks0L0hgznZLG9nzXb8vZ0rRvqNvAcgAp84y7Mwkgw=
gonum.org/v1/gonum v0.7.0/go.mod h1:L02bwd0sqlsvRv41G7wGWFCsVNZFv/k1xzGIxeANHGM=
R

### Does this issue reproduce with the current master?
I haven't tried it.

### PR DESCRIPTION
**Pull request by romanwerpachowski**
Fixes #1377.

Please take a look.

### CODE REVIEW

File: stat/distuv/beta.go [hunk 1]
```diff
@@ -71,7 +71,19 @@ func (b Beta) LogProb(x float64) float64 {
 	lab, _ := math.Lgamma(b.Alpha + b.Beta)
 	la, _ := math.Lgamma(b.Alpha)
 	lb, _ := math.Lgamma(b.Beta)
-	return lab - la - lb + (b.Alpha-1)*math.Log(x) + (b.Beta-1)*math.Log(1-x)
+	var lx float64
+	if b.Alpha != 1 {
+		lx = (b.Alpha - 1) * math.Log(x)
+	} else {
+		lx = 0
```
Comment (kortschak) [inline stat/distuv/beta.go:78]: This is not necessary since `lx` is initialised to 0.

File: stat/distuv/beta.go [hunk 2]
```diff
@@ -71,7 +71,19 @@ func (b Beta) LogProb(x float64) float64 {
...
+	}
+	var l1mx float64
+	if b.Beta != 1 {
+		l1mx = (b.Beta - 1) * math.Log(1-x)
+	} else {
+		l1mx = 0
```
Comment (kortschak) [inline stat/distuv/beta.go:84]: Same here.

File: stat/distuv/beta_test.go [hunk 3]
```diff
@@ -148,3 +148,26 @@ func TestBetaMode(t *testing.T) {
 		}
 	}
 }
+
+func TestBetaIssue1377(t *testing.T) {
```
Comment (kortschak) [inline stat/distuv/beta_test.go:152]: Please add a link to the github issue.

**Comment by romanwerpachowski** [comment]
PTAL.

**Pull request merged**